### DISCUSSION TITLE
webapi has "stopped updating" [root cause -> Senec Webservice provide lower values as before]

### DISCUSSION THREAD

**Discussion by Disincentivato** [marq24/ha-senec-v3#48] [Q&A] [answered]
Since this morning the webapi have no longer been updated. What could it be?
![Immagine 2023-12-05 163738](https://github.com/marq24/ha-senec-v3/assets/90531670/a291d8ac-4403-452d-9ba4-141d31e4186b)

**Comment by BMeyendriesch** [comment]
Same problem here.

**Comment by marq24** [comment]
What when you access the webportal manually ? Do you get updated data?  - also you might like to enable debug log and check for possible messages from the integration - and if you have not done it already - restart your HA...

At least for me the API works flawless @ home...

![image](https://github.com/marq24/ha-senec-v3/assets/8012359/68529fa6-5691-4ce1-adfe-ba5751f6f474)

**Reply by BMeyendriesch** [reply]
The SENEC web portal provides continuous values. However, these are smaller than the last values read out by ha-senec-v3. Therefore, your homeassistant integration logically does not provide me with any new measured values. Thank you very much, SENEC!

```
2023-12-06 10:09:35.581 DEBUG (MainThread) [custom_components.senec] Finished fetching senec data in 3.416 seconds (success: True)
2023-12-06 10:09:35.581 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key consumption_total - prev:17571.45 new: 17520.02
2023-12-06 10:09:35.582 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key powergenerated_total - prev:28889.53 new: 28824.49
2023-12-06 10:09:35.582 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key accuimport_total - prev:3929.25 new: 3911.41
2023-12-06 10:09:35.582 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key accuexport_total - prev:4127.34 new: 4110.56
2023-12-06 10:09:35.582 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key gridimport_total - prev:7910.75 new: 7898.55
2023-12-06 10:09:35.583 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key gridexport_total - prev:19206.39 new: 19179.45
```

**Comment by neubi4** [comment]
My HA also has this problem, since this exactly same time. Restarting HA did not help. There are also no errors of this integration at HA log.

Data in HA differs from mein-senec, but it seems the numbers in HA a bigger than mein-senec.

Also, some data are still updated, for example current battery percentage.

![Screenshot_20231205-223311](https://github.com/marq24/ha-senec-v3/assets/1274627/907c07f4-f58f-4d11-8f45-c962ee8fe07c)
![Screenshot_20231205-223251](https://github.com/marq24/ha-senec-v3/assets/1274627/9b4f5176-2b0f-44c8-8440-66be754112d1)

**Comment by marq24** [comment, marked as answer]
So when the values you see in the webportal are LOWER, that the data recorded in HA, then you are running in #44. Senec have a issue with providing the correct data - not only since today! 

So instead that this HA integration will fetch this false data, the latest build will check, if the values are increasing - if not you will see a corresponding entry in the LOG (if debug is enabled).

Would be cool, if somebody of you could confirm, that you have the log entry (and that the "Senec is providing false data" is the root cause of this) - TIA

**Reply by neubi4** [reply]
I activated the debug log, and you are right:
```
2023-12-06 09:57:28.866 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key accuimport_total - prev:2652.8 new: 2647.04
2023-12-06 09:57:28.866 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key accuexport_total - prev:2790.77 new: 2786.48
2023-12-06 09:57:28.866 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key gridimport_total - prev:2710.41 new: 2700.47
2023-12-06 09:57:28.866 DEBUG (MainThread) [custom_components.senec.sensor] Thanks for nothing Senec! prev>new for key gridexport_total - prev:10319.07 new: 10316.56
```

Since this happened at the same time as @Disincentivato, i assume Senec "corrected" some values in their database or something.
Even more funny, Senec App shows Total Grid Import at 2698 kWh. Another different value...

So this "Issue" fixes itself when Senecs Web values are higher then the HA values again. Thank you @marq24 

Thanks for nothing Senec!

**Comment by Disincentivato** [comment]
I think that the best way in order to get the various kWh is to remove the WEBAPI integration and create some **Riemann Sum Integral** from the helper page.
I've been using this value since there was a similar problem with webapi and I have to say that the difference is negligible. The data is linked to consumption taken from local integration so I don't see many negative sides.
I already mentioned it in the issue 44 post, let me know what you think.
_*In any case this morning the total consumption from the webapi is obviously stuck at 0._

![image](https://github.com/marq24/ha-senec-v3/assets/90531670/25a88680-6928-4383-8ae2-e70b00c013fc)

**Reply by BMeyendriesch** [reply]
I have switchend from WebAPI to PowerCalc energy integrator. Works fine . . .
PowerCalc: Home Assistant Virtual Power Sensors

**Reply by bobthebuilder4711** [reply]
@Disincentivato This can work well but it can also drift a lot. Especially if the senec storage simply refuses new http connections for some time, which happens quite often for me.

But if you just want a ballpark number and don't care about a few hundred kWh per year then this is likely the better solution, yes.

**Reply by BMeyendriesch** [reply]
Do you think Riemann Sum Integral works better than PowerCalc?

**Reply by bobthebuilder4711** [reply]
https://homeassistant-powercalc.readthedocs.io/en/latest/sensor-types/energy-sensor.html
>Powercalc can automatically create energy sensors for your virtual power meters. **It uses the Riemann Sum helper for that.**

Means it's literally the same thing.

**Comment by AndreWohnsland** [comment]
Thanks again for the investigation. I can confirm the same behaviour since last evening, Senec seemed to adjust all values to the drifting (maybe now correct) value, or just some other bug. Anyways it is causing no increase in the values, as expected. I hope the values increase enough in some time to hit the old value and work properly again. Otherwise using the Riemann integration is the best bet, even if it will be off by some amount (I also lose the connection sometimes).

**Comment by bobthebuilder4711** [comment]
I guess the best option while using the integration would be to have some kind of button with which you can force the integration to use the current values from mein-senec.de and ignore the "only rising values" rule and then keep going from there.

I did just that with my own script, which resulted in this: 
![image](https://github.com/marq24/ha-senec-v3/assets/95188461/b06a8eb4-97bb-4bf8-9652-250d52c06c07)

And then I can go to the developer tools and fix that by setting the value at 15:20 to "0":
![image](https://github.com/marq24/ha-senec-v3/assets/95188461/7f7e18d6-0695-4c53-8a34-6bba734b631d)

This won't change the values that HA has in its "short term memory" that is kept for two weeks and is then deleted anyway. But it fixes the statistics the recorder creates for every 5 minutes and later for every hour.

**Comment by loeffelpan** [comment]
Stopping HA and deleting the whole database will also do the trick, as new api values will become new total sensors values.

Just energy data and all state history will get lost.
